tutorial.md

```diff
@@ -0,0 +1,212 @@
+
+Some Google Cloud operations may take a long time to complete. Here's how to handle them:
+
+```rust
+use google_cloud_speech_v2::client::SpeechClient;
+use google_cloud_gax::retry::ExponentialBackoff;
+use std::time::Duration;
+
+async fn handle_long_running_operation(
+    project_id: &str
+) -> Result<(), Box<dyn std::error::Error>> {
+    // Initialize client with custom polling policy
+    let backoff = ExponentialBackoff::new(
+        Duration::from_secs(1),    // Initial delay
+        Duration::from_secs(60),   // Maximum delay
+        1.5,                       // Multiplier
+    );
+    
+    let client = SpeechClient::builder()
+        .with_polling_backoff_policy(backoff)
+        .build()
+        .await?;
+    
+    // Start a long-running operation
+    let operation = client
+        .create_recognizer()
+        .set_parent(format!("projects/{}/locations/global", project_id))
+        .set_recognizer_id("my-recognizer")
+        .send()
+        .await?;
+    
+    // Wait for completion
+    let result = operation.await_complete().await?;
+    
+    Ok(())
+}
+```
+
+### Error handling
+
+Handle errors appropriately to ensure your application is robust:
+
+```rust
+use google_cloud_gax::error::{Code, ServiceError};
+
+async fn robust_error_handling(
+    project_id: &str
+) -> Result<(), Box<dyn std::error::Error>> {
+    let client = match SecretManagerServiceClient::new().await {
+        Ok(client) => client,
+        Err(e) => {
+            eprintln!("Failed to initialize client: {}", e);
+            return Err(e.into());
+        }
+    };
+
+    let secret_name = format!("projects/{}/secrets/my-secret", project_id);
+    
+    match client.get_secret().set_name(secret_name).send().await {
+        Ok(secret) => {
+            println!("Retrieved secret: {}", secret.name);
+            Ok(())
+        }
+        Err(e) => {
+            if let Some(status) = e.downcast_ref::<ServiceError>() {
+                match status.code() {
+                    Code::NotFound => {
+                        println!("Secret not found");
+                        Ok(())
+                    }
+                    Code::PermissionDenied => {
+                        eprintln!("Permission denied");
+                        Err(e)
+                    }
+                    _ => {
+                        eprintln!("Service error: {}", status);
+                        Err(e)
+                    }
+                }
+            } else {
+                eprintln!("Unknown error: {}", e);
+                Err(e)
+            }
+        }
+    }
+}
+```
+
+## Troubleshooting
+
+### Common issues
+
+1. **Authentication errors**
+   - Ensure you've run `gcloud auth application-default login`
+   - Verify your service account has the necessary permissions
+
+2. **API not enabled**
+   - Enable the required APIs in your project:
+     ```bash
+     gcloud services enable secretmanager.googleapis.com
```

Comment:
You are missing some APIs used above, like `speech.googleapis.com`. Keeping the tutorial focused on one topic makes it easier to catch these problems.